emcc.py

```diff
@@ -1386,7 +1386,7 @@ def compile_source_file(i, input_file):
        not shared.Settings.ONLY_MY_CODE and \
        not shared.Settings.SIDE_MODULE: # shared libraries/side modules link no C libraries, need them in parent
       extra_files_to_link = system_libs.get_ports(shared.Settings)
-      extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout, stderr, forced=forced_stdlibs)
+      extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout_=None, stderr_=None, forced=forced_stdlibs)
```

Comment:
Btw @kripken: it looks like we have had some cross-platform behavior discrepancy here, as on Linux and OS X any build warnings from system libs did not show up, so e.g. #4393 did not manifest on either of those OSes. On Windows the outputs would get printed even if they were PIPEd here. That difference doesn't matter much though, because we should make sure our system libs never emit any build warnings, so we can just print those out (to make it easier to work with updating system libs), and added `-Werror` requirement to the build so that those won't go uncaught.